packages/varnish/src/header/Header.tsx

```diff
@@ -0,0 +1,223 @@
+// levels when implemnting something more custom. This way the user can do everything
+// with the <Header.*> components, and doesn't need to use the antd ones.
+// const Container = Layout.Header;
+
+// const Content = styled(VContent)`
+//     display: flex;
+//     align-items: center;
+//     padding-top: 0;
+//     padding-bottom: 0;
+// `;
+
+const AppName = ({ children }: { children: string }) => {
+    if (children.length > 24) {
+        console.warn('The contents of `<AppName>` should be fewer then 25 characters');
+    }
+    return <AppNameText>{children}</AppNameText>;
+};
+
+const AppNameText = styled.h3`
+    margin: 0;
+    padding: 0;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    font-size: ${pxToRem(30)};
+
+    @media (max-width: 480px) {
+        font-size: 24px;
+    }
+`;
+
+const AppTagline = styled.h4`
+    margin: 0;
+    padding: 0;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    font-size: ${pxToRem(24)};
+
+    @media (max-width: 8px) {
+        display: none;
+    }
+`;
+
+const LogoGrid = styled.span`
+    display: grid;
+    grid-template-columns: repeat(4, auto);
+    align-items: center;
+    gap: 8px;
+
+    &:hover ${AppNameText} {
+        text-decoration: none;
+        display: inline-block;
+    }
+`;
+
+const DecorationlessAnchor = styled(Link)`
+    &&:hover {
+        text-decoration: none;
+        color: ${varnishColor.N9.toString()};
+    }
+    text-decoration: none;
+    color: ${varnishColor.N9.toString()};
+`;
+
+// Wraps the header logo with an ahref if href is passed in
+const ConditionalAnchorWrapper = (props: { href?: string; children: React.ReactElement }) => {
+    return props.href ? (
+        <DecorationlessAnchor href={props.href} rel="noopener noreferrer">
+            {props.children}
+        </DecorationlessAnchor>
+    ) : (
+        props.children
+    );
+};
+
+const Logo = (props: {
+    label?: React.ReactElement;
+    tagline?: React.ReactElement;
+    separator?: React.ReactElement;
+    href?: string;
+    children?: React.ReactElement;
+    className?: string;
+}) => {
+    return (
+        <ConditionalAnchorWrapper href={props.href}>
+            <LogoGrid className={props.className}>
+                {props.children}
+                {props.label}
+                {props.label && props.tagline ? (
+                    props.separator ? (
+                        props.separator
+                    ) : (
+                        <AppTagline>&mdash;</AppTagline>
+                    )
+                ) : null}
+                {props.tagline}
+            </LogoGrid>
+        </ConditionalAnchorWrapper>
+    );
+};
+
+interface ColumnsProps {
+    columns?: string;
+}
+
+const MaxWidthDiv = styled.div`
+    width: 100%;
+`;
+
+const StyledAppBar = styled(AppBar)`
+    background-color: white;
+    color: ${varnishColor.N9.toString()};
+    size: 22pt;
+    top: 0;
+    z-index: 950;
```

Comment:
iz z index needed on AppBar?

Reply:
whoops no good catch, appbar defaults to a z-index of >1000